cloudinit/sources/DataSourceVMware.py

```diff
@@ -0,0 +1,780 @@
+        return None
+    enc_type = guestinfo_envvar_get_value(key + ".encoding")
+    return decode("guestinfo." + key, enc_type, val)
+
+
+def guestinfo_envvar_get_value(key):
+    env_key = ("vmx.guestinfo." + key).upper().replace(".", "_", -1)
+    return handle_returned_guestinfo_val(key, os.environ.get(env_key, ""))
+
+
+def guestinfo(key, vmware_rpctool=VMWARE_RPCTOOL):
+    """
+    guestinfo returns the guestinfo value for the provided key, decoding
+    the value when required
+    """
+    val = guestinfo_get_value(key, vmware_rpctool)
+    if not val:
+        return None
+    enc_type = guestinfo_get_value(key + ".encoding", vmware_rpctool)
+    return decode("guestinfo." + key, enc_type, val)
+
+
+def guestinfo_get_value(key, vmware_rpctool=VMWARE_RPCTOOL):
+    """
+    Returns a guestinfo value for the specified key.
+    """
+    LOG.debug("Getting guestinfo value for key %s", key)
+
+    try:
+        (stdout, stderr) = subp([vmware_rpctool, "info-get guestinfo." + key])
+        if stderr == NOVAL:
+            LOG.debug("No value found for key %s", key)
+        elif not stdout:
+            LOG.error("Failed to get guestinfo value for key %s", key)
+        return handle_returned_guestinfo_val(key, stdout)
+    except ProcessExecutionError as error:
+        if error.stderr == NOVAL:
+            LOG.debug("No value found for key %s", key)
+        else:
+            util.logexc(
+                LOG,
+                "Failed to get guestinfo value for key %s: %s",
+                key,
+                error,
+            )
+    except Exception:
+        util.logexc(
+            LOG,
+            "Unexpected error while trying to get "
+            + "guestinfo value for key %s",
+            key,
+        )
+
+    return None
+
+
+def guestinfo_set_value(key, value, vmware_rpctool=VMWARE_RPCTOOL):
+    """
+    Sets a guestinfo value for the specified key. Set value to an empty string
+    to clear an existing guestinfo key.
+    """
+
+    # If value is an empty string then set it to a single space as it is not
+    # possible to set a guestinfo key to an empty string. Setting a guestinfo
+    # key to a single space is as close as it gets to clearing an existing
+    # guestinfo key.
+    if value == "":
+        value = " "
+
+    LOG.debug("Setting guestinfo key=%s to value=%s", key, value)
+
+    try:
+        subp([vmware_rpctool, ("info-set guestinfo.%s %s" % (key, value))])
+        return True
+    except ProcessExecutionError as error:
+        util.logexc(
+            LOG,
+            "Failed to set guestinfo key=%s to value=%s: %s",
+            key,
+            value,
+            error,
+        )
+    except Exception:
+        util.logexc(
+            LOG,
+            "Unexpected error while trying to set "
+            + "guestinfo key=%s to value=%s",
+            key,
+            value,
+        )
+
+    return None
+
+
+def guestinfo_clear_keys(keys, vmware_rpctool=VMWARE_RPCTOOL):
```

Comment:
bikeshed function rename just because we use "redact" in a number of places in cloudinit and it would be nice to have greps find them all in the future in case we standardize redact operations for all datasources, logging, instance data etc.
 
```suggestion
def guestinfo_redact_keys(keys, vmware_rpctool=VMWARE_RPCTOOL):
```

Reply:
Sure thing!